platform-api/che-core-api-builder/src/main/java/org/eclipse/che/api/builder/internal/Builder.java

```diff
@@ -444,6 +444,7 @@ public Boolean call() throws Exception {
                 try {
                     ProcessBuilder processBuilder = new ProcessBuilder().command(commandLine.toShellCommand()).directory(
                             configuration.getWorkDir()).redirectErrorStream(true);
+                    modifyCommandEnvironment(processBuilder.environment(), configuration);
                     Process process = processBuilder.start();
```

Comment:
I suggest using java 8's built-in waitFor(time, timeUnit) instead of using ad-hoc polling code. this areas is a candidate for several poll requests I sugess